src/library/basesqltablemodel.h

```diff
@@ -139,6 +139,23 @@ class BaseSqlTableModel : public QAbstractTableModel, public TrackModel {
         }
         int m_column;
         Qt::SortOrder m_order;
+        
+        friend QTextStream& operator >> (QTextStream &in, SortColumn &sc) {
```

Comment:
Ok. that sounds reasonable.

> Do you mean that the header data already contains sorting state so there's the possibility for this PR to store data conflicting with the header state? 

No, because we need here the sorting hierarchy.